pandas/core/describe.py

```diff
@@ -240,7 +240,12 @@ def describe_numeric_1d(series: Series, percentiles: Sequence[float]) -> Series:
         + series.quantile(percentiles).tolist()
         + [series.max()]
     )
-    return Series(d, index=stat_index, name=series.name)
+
+    result = Series(d, index=stat_index, name=series.name)
+    if isinstance(d[1], float):
```

Comment:
Can we determine the target dtype based on the input dtype? Like when we have numeric cast to float, else keep?

Reply:
Thanks  - I was worried about doing this because of object dtype with numeric data, but I see now object dtype takes a different path.